src/scanner.cc

```diff
@@ -88,7 +88,7 @@ struct Scanner {
 
   bool scan_heredoc_end_identifier(TSLexer *lexer) {
     current_leading_word.clear();
-    while (iswalpha(lexer->lookahead)) {
+    while ((iswalpha(lexer->lookahead) || iswspace(lexer->lookahead)) && current_leading_word.length() < heredoc_delimiter.length()) {
```

Comment:
I think the logic you added about `heredoc_delimiter.length()` is good, but I think we also need to terminate this loop when we see a newline character. In fact, we could probably remove the `iswalpha` check in favor of that:

```c
// Scan the first 'n' characters on this line, to see if they match the heredoc delimiter
while (
  lexer->lookahead != '\0' &&
  lexer->lookahead != '\n' &&
  current_leading_word.length() < heredoc_delimiter.length()
) {
  // ...
}
```

Otherwise, if our heredoc contains some newlines, we might consume the last line of the heredoc, then consume the newline character and *then* start consuming the actual heredoc ending, all as part of one call to `scan_heredoc_end_identifier`.